server/src/main/java/io/crate/execution/engine/sort/Sort.java

```diff
@@ -44,26 +52,41 @@ public static <T> CompletableFuture<List<T>> parallelSort(List<T> list,
         int itemsPerThread = list.size() / Math.max(1, numAvailableThreads);
         if (numAvailableThreads <= 1 || itemsPerThread < minItemsPerThread) {
             list.sort(comparator);
+            allocateBytes.accept(sortOverheadBytes(list.size()));
```

Comment:
sort creates an intermediate array `Object[] a = this.toArray();`

Reply:
We should account at best before the memory is used, so in this case before the `list.sort`. 
Additionally, only the default `sort` impl of the `List` is converting to an array, while e.g. the `ArrayList.sort` isn't, it uses `Arrays.sort` which has different memory consumption according to the javadocs.
Anyway, I'm fine with the potential over-accounting, but lets do it before the usage.